### PR TITLE
adding common test framework basics

### PR DESCRIPTION
**Pull request by jaypipes**
Pulls code from the main community repo for the framework for functional
integration testing in Python. Adds some basic infrastructure and docs
for setting up a local development environment (for developing
test-infra itself).

### CODE REVIEW

**Comment by jaypipes** [comment]
@A-Hilaly @RedbackThomson @vijtrip2 OK, so this PR just pulls in some very basic infrastructure from the community repo for a testing framework. Doesn't include the bootstrap/cleanup/conftest.py files yet so isn't functional but this is just to get ideas on splitting out the framework that we can then use in individual service controller repositories (as a Python requirement/dependency)

File: NOTICE [hunk 1]
```diff
@@ -1 +1 @@
-Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (RedbackThomson) [inline NOTICE:1]: Is ... this up to date?

File: docs/setup.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+```bash
+python3 -m pip install --user pipx
+python3 -m pipx ensurepath
+```
+
+Install the `pipenv` tool for creating Python3 virtual environments:
+
+```bash
+pipx install pipenv
```
Comment (RedbackThomson) [inline docs/setup.md:22]: Using `pipx` to install `pipenv`? I've never heard about this before

File: NOTICE (hunk 1 above)
Comment (jaypipes) [reply NOTICE:1]: clearly not :) I really should just get tid of these NOTICE files entirely. They are pointless. This was just copy/pasta from the original template repository...

File: docs/setup.md (hunk 2 above)
Comment (jaypipes) [reply docs/setup.md:22]: The benefits of pipx I believe will become super evident as we end up having a common test-infra Python framework and CLI application along with e2e tests and potentially Python CLI applications in lots of the service controller repositories. Having pipx's ease of dealing with Python code that is arranged in local Git repositories will be invaluable I think.

**Review by vijtrip2 — APPROVED** [review]
LGTM

**Review by A-Hilaly — APPROVED** [review]
👍 🚢 it!

**Review by RedbackThomson — APPROVED** [review]

**Pull request merged**